Kudu.Core/Helpers/PostDeploymentHelper.cs

```diff
@@ -740,5 +741,33 @@ public static async Task UpdateSiteVersion(ZipDeploymentInfo deploymentInfo, IEn
                 await FileSystemHelpers.WriteAllTextToFileAsync(siteVersionPath, deploymentInfo.ZipName);
             }
         }
+
+        public static void RemoveAppOfflineIfLeft(IEnvironment environment, IOperationLock deploymentLock, ITracer tracer)
+        {
+            var appOfflineFile = Path.Combine(environment.WebRootPath, Constants.AppOfflineFileName);
+            if (FileSystemHelpers.FileExists(appOfflineFile))
+            {
+                if (deploymentLock.IsHeld)
+                {
+                    tracer.Trace($"Deployment lock is held, will not remove {Constants.AppOfflineFileName}");
+                    return;
+                }
+
+                try
```

Comment:
I would try/catch the whole thing to be saved.

Reply:
Good point! I meant that have the callers use "SafeOperation". But catching the whole thing and logging is probably a better option.